packages/binding-coap/src/coap.ts

```diff
@@ -64,3 +76,22 @@ export function isValidCoapMethod(methodName: CoapMethodName): methodName is Coa
 export function isSupportedCoapMethod(methodName: CoapMethodName): methodName is CoapMethodName {
     return ["GET", "POST", "PUT", "DELETE"].includes(methodName);
 }
+
+/**
+ * Encodes block size values for blockwise transfer to CoAP-specific values.
+ *
+ * Maps the values 16, 32, 64, 128, 256, 512, and 1024 to 0, 1, 2, 3, 4, 5, and 6.
+ * Other values will be rounded to the nearest value accepted by CoAP.
+ *
+ * @param blockSize The original block size value.
+ * @returns A mapped block size value usable by CoAP.
+ */
+export function blockSizeToOptionValue(blockSize: BlockSize): BlockSizeOptionValue {
+    // TODO: Check if there is a more elegant solution.
+    const blockSizeOptionValue = Math.floor(Math.sqrt(blockSize) - 4);
```

Comment:
[Math.log2](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log2)?

Reply:
Thanks, adjusted in 501929a0bd76c10d66f3609b35ea70c8689f04d2 :)

Reply:
You're right, that would actually be unintuitive. I converted the function's implementation in 26f53d026af6ed72e3e6dd60a2ecccbf4fe7fad6 to a switch statement, capturing only the allowed values. Hope that it hasn't become a bit too verbose this way.